R/merge.R

```diff
@@ -11,9 +11,21 @@ merge.data.table = function(x, y, by = NULL, by.x = NULL, by.y = NULL, all = FAL
       by = key(x)
     }
   }
-  if ((x0 <- length(x)==0L) | (y0 <- length(y)==0L)) warning("You are trying to join data.tables where ", if(x0 & y0) "'x' and 'y' arguments are" else if(x0 & !y0) "'x' argument is" else if(!x0 & y0) "'y' argument is", " 0 columns data.table.")
-  if (any(duplicated(names(x)))) stop("x has some duplicated column name(s): ",paste(names(x)[duplicated(names(x))],collapse=","),". Please remove or rename the duplicate(s) and try again.")
-  if (any(duplicated(names(y)))) stop("y has some duplicated column name(s): ",paste(names(y)[duplicated(names(y))],collapse=","),". Please remove or rename the duplicate(s) and try again.")
+  x0 = length(x)==0L
+  y0 = length(y)==0L
+  if (x0 || y0) warning(sprintf(
```

Comment:
Any idea why this would be a warning and not an error by the way?? How can we merge when `x` or `y` have no columns?? Certainly, the tests of this `warning` end up failing a few lines later.

Should we just change this to `stop`?

Reply:
I wouldn't drive that much change of code just because of the linter.